step-plans/step-plans-yaml-parser/src/main/java/step/plans/parser/yaml/rules/DataSourceFieldsYamlHelper.java

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
```

Comment:
I assume these changes are related to the support of For and ForEach and have been reviewed in a dedicated PR. Not reviewing them here

Reply:
@jeromecomte 
Can I reuse the original branch number (SED-2516) for the separate PR to be merged into SED-2518? Or it better to create another one?

Reply:
@iegorov777 Haven't you already merged the changes of SED-2516 into this branch (SED-2555)? are there other changes to be merged to the epic branch?

Reply:
@iegorov777 
All changes from SED-2516 are already merged into SED-2555, but I've prepared the separate PR for SED-2516 just to split changes related to the SED-2516 for separate review (if required).